File: 100_doors.js
/* a program that opens and closes doors
https://www.freecodecamp.org/learn/coding-interview-prep/rosetta-code/100-doors
*/

function getFinalOpenedDoors(numDoors) {
    const doors = [];
    for (let i = 1; i <= numDoors; i++) {
      for (let num = 1; num <= numDoors; num++) {
        if (doors.includes(num) == false && num % i == 0) {
          doors.push(num);
        } else if (doors.includes(num) && num % i == 0) {
            doors.splice(doors.indexOf(num), 1);

        } else {
          continue;
        }
      } 
    } return doors;
    }
console.log(getFinalOpenedDoors(100));

File: 100_doors.py
# A program that opens and closes doors
# https://www.freecodecamp.org/learn/coding-interview-prep/rosetta-code/100-doors


doors = []
i = 1
for i in range(1, 101):
    for num in range(1, 101):
        if num not in doors and num % i == 0:
            doors.append(num)
            doors.sort()
        elif num in doors and num % i == 0:
            doors.remove(num)
            doors.sort()
        else:
            continue
    print(f'{i}:{doors}\n')

